src/python/pants/invalidation/cache_manager.py

```diff
@@ -168,42 +186,34 @@ def _results_dir_path(self, root_dir, key, stable):
         self._STABLE_DIR_NAME if stable else sha1(key.hash).hexdigest()[:12]
     )
 
-  def create_results_dir(self, root_dir, allow_incremental):
-    """Ensures that a results_dir exists under the given root_dir for this versioned target.
+  def create_results_dir(self, root_dir):
+    """Ensure that the empty results directory and a stable symlink exist for these versioned targets."""
+    self._current_results_dir = self._results_dir_path(root_dir, self.cache_key, stable=False)
+    self._results_dir = self._results_dir_path(root_dir, self.cache_key, stable=True)
 
-    If incremental=True, attempts to clone the results_dir for the previous version of this target
-    to the new results dir. Otherwise, simply ensures that the results dir exists.
-    """
-    # Generate unique and stable directory paths for this cache key.
-    current_dir = self._results_dir_path(root_dir, self.cache_key, stable=False)
-    self._current_results_dir = current_dir
-    stable_dir = self._results_dir_path(root_dir, self.cache_key, stable=True)
-    self._results_dir = stable_dir
-    if self.valid:
-      # If the target is valid, both directories can be assumed to exist.
-      return
-
-    # Clone from the previous results_dir if incremental, or initialize.
-    previous_dir = self._use_previous_dir(allow_incremental, root_dir, current_dir)
-    if previous_dir is not None:
-      self.is_incremental = True
-      self._previous_results_dir = previous_dir
-      shutil.copytree(previous_dir, current_dir)
-    else:
-      safe_mkdir(current_dir)
+    if not self.valid:
+      # Clean the workspace for invalid vts.
+      safe_mkdir(self._current_results_dir, clean=True)
+      relative_symlink(self._current_results_dir, self._results_dir)
+    self.ensure_legal()
 
-    # Finally, create the stable symlink.
-    relative_symlink(current_dir, stable_dir)
+  def copy_previous_results(self, root_dir):
```

Comment:
Might make better sense to move to Task honestly - it is weird that the cache manager manages that part.

I have a followup that refactors cache_manager, will consider doing that there.